### ISSUE TITLE
$cordovaGeolocation - Android error message : Only secure origins are allowed

### ISSUE THREAD

**Issue by TuNi54** [driftyco/ng-cordova#1413]
Hello,

I'm using the "$cordovaGeolocation" plugin, everything is OK on iOS but on Android (ionic run android --device -c -l) I receive this following error message :

"Only secure origins are allowed (see: https://goo.gl/Y0ZkNV)."

Is it because i'm running the reload server in http ?

```
Running live reload server: http://192.168.1.12:35729
Watching: www/**/*, !www/lib/**/*
√ Running dev server: http://192.168.1.12:8100
```

What's the solution please ?

Thank you!

**Comment by philonor** [comment]
Your guess is correct, accessing the user's location is deprecated with unsecure (HTTP) context. See https://www.chromium.org/Home/chromium-security/prefer-secure-origins-for-powerful-new-features.

Running your dev server on `localhost` instead should work.
> (*, localhost, *)

Did you try to setup a [HTTPS proxy](https://www.petrelli.biz/2017/07/12/ionic-serve-how-to-connect-over-https-to-use-reserved-html5-methods-such-as-geolocation/)?
- Run your ionic project without SSL using `ionic serve` at port `8100`
- Setup HTTPS proxy at port `8101` to be able to use privacy-critical functions like geolocation
- Connect with your mobile device at `https://proxy-ip:8101`

It seems that they are [targeting SSL support](https://github.com/ionic-team/ionic-cli/issues/2671) for the Ionic CLI in a future version. But until then you can use the mentioned workaround.

**Comment by geninsin** [comment]
This worked for me, @TuNi54 
https://forum.ionicframework.com/t/ionic2-geolocation-only-secure-origins-are-allowed-error/61333/6

**Comment by TroubadourFabiolas** [comment]
Hi @TuNi54 @philonor @geninsin , thanks for yours answers,

**Same issues here :** 
https://forum.ionicframework.com/t/ionic-geolocation-only-secure-origins-are-allowed-error/61333/11
https://stackoverflow.com/questions/45647202/how-to-solve-only-secure-origins-are-allowed

1) Geolocation works with this command : ionic cordova run android

2) **How can we make work Geolocation with livereload ?**

2.1)  I try this without success : ionic cordova run android -c-l-s --address MY_IP_ADRESS --dev
=> error : "Only secure origins are allowed (see: https://goo.gl/Y0ZkNV)."

2.2)  I try this without success : ionic cordova run android -c-l-s --address localhost --dev
=> on Terminal : [WARN] Your device or emulator may not be able to access localhost.
Ensure you have proper port forwarding setup from your device to your computer.
=> on Device : Application Error : the connection to the server was unsuccessful. (http://localhost:8100)

**My env**

```
cli packages: (/usr/local/lib/node_modules)

    @ionic/cli-utils  : 1.9.2
    ionic (Ionic CLI) : 3.9.2

global packages:

    Cordova CLI : 7.0.1 

local packages:

    Cordova Platforms : android 6.2.3 ios 4.4.0
    Ionic Framework   : ionic1 1.2.4

System:

    Android SDK Tools : 25.2.3
    ios-deploy        : 1.8.5 
    ios-sim           : 5.0.8 
    Node              : v8.4.0
    npm               : 5.3.0 
    OS                : OS X El Capitan
    Xcode             : Xcode 7.3.1 Build version 7D1014 
```

**Any idea ?**

**Thank you**

**Comment by TroubadourFabiolas** [comment]
no, this issue is not closed ?

**Comment by morphatic** [comment]
I can confirm what @TroubadourFabiolas  reported. Although I can test functionality that relies on native geolocation using `ionic cordova run android`, because the `--livereload` and `--console` functionality requires running a server that can be accessed from the dev machine, NOT allowing connections to the test server via SSL means you can't use the native geolocation features.

Is there any way to run the debugging server for `--livereload` and/or `--console` from HTTPS?

**Comment by ythalorossy** [comment]
I confirm too. Using the command `ionic cordova run android --livereload` we can simulate this problem.

**Comment by morphatic** [comment]
I believe the livereload server needs to default to using SSL in order to allow you to use it for testing Geolocation.

**Comment by fdambrosio** [comment]
I have the same problem with --livereload

**Comment by albert-92** [comment]
Me too with `--livereload`.
Is there any workaround?

**Comment by frankcalpito** [comment]
Same here pls default to SSL

**Comment by fdambrosio** [comment]
what you mean @juncalpito ?

**Comment by frankcalpito** [comment]
I just seconded what morphatic said about the livereload server needs to default to using SSL, so that we won't have any problems with requesting from insecure origins.

**Comment by CD-UNCC** [comment]
With or without live reload I cannot use the ionic dev app and the geolocation api together. Watching for a fix.

**Comment by steinjonathan** [comment]
The same for me. The app crashes suddenly.
There is a solution?

**Comment by matheuscaldasrj** [comment]
With or without live reload option I am still getting the same error 👎

**Comment by steinjonathan** [comment]
@matheuscaldasrj I removed ionic cordova android, deleted the node_modules folder and plugins folder, installed ionic cordova android and then ionic cordova run android.

**Comment by lefrankleal** [comment]
please revive!!! any solution for this?

**Comment by fdambrosio** [comment]
there's a workaround?

**Comment by steinjonathan** [comment]
@lefrankleal @fdambrosio I removed ionic cordova android, deleted the node_modules folder and plugins folder, installed ionic cordova android and then ionic cordova run android.

**Comment by Diosan** [comment]
@steinjonathan Yes that is the point. It works with run . What we want is it to work with --livereload  or -l option like this below

`ionic cordova run android -l`

**Comment by darkguy2008** [comment]
Still 2018 and this issue persists

**Comment by samaranin** [comment]
So, no solution for this?

**Comment by zbahadir** [comment]
> Still 2018 and this issue persists

It's almost 2019.

**Comment by gortok** [comment]
@zbahadir This issue is never getting fixed by the ionic team; this library is no longer maintained.  https://github.com/ionic-team/ng-cordova/blob/master/README.md

**Comment by dmsanchez86** [comment]
Issue persists 2019

**Comment by abalad** [comment]
same here

**Comment by mfouadegy** [comment]
same here :(

**Comment by fdambrosio** [comment]
any news? :-(

**Comment by corncobb** [comment]
Almost 2020 and still no fix?

**Comment by amansheaven** [comment]
In 2020 still no fix :(

**Comment by gortok** [comment]
Politely: there will never be a fix. This is open source code; feel free to fork and maintain what you need.

At a minimum you’ll need to look at the code this library wraps and see if the fix is there. If so, fix it there and then this will work; if not you may need to fork and fix the base library as well as this wrapper.

**Comment by darkguy2008** [comment]
> Politely: there will never be a fix. This is open source code; feel free to fork and maintain what you need.
> 
> At a minimum you’ll need to look at the code this library wraps and see if the fix is there. If so, fix it there and then this will work; if not you may need to fork and fix the base library as well as this wrapper.

You guys never disappoint, I didn't expect more from the ionic team itself. Way to go!

`</sarcasm>`

**Comment by corncobb** [comment]
I only encountered this issue when running on a simulator. Not a big deal to me anymore.

**Comment by alyleite** [comment]
any news?

**Comment by toshkoevo** [comment]
Broken on a real android device with livereload in 2021.
